pkg/osquery/enrollment_details.go

```diff
@@ -0,0 +1,141 @@
+package osquery
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/pkg/agent"
+	"github.com/kolide/launcher/pkg/osquery/runsimple"
+	"github.com/kolide/launcher/pkg/service"
+	"github.com/pkg/errors"
+)
+
+func getEnrollDetails(ctx context.Context, osquerydPath string) (service.EnrollmentDetails, error) {
+	var details service.EnrollmentDetails
+
+	// To facilitate manual testing around missing enrollment details,
+	// there is a environmental variable to trigger the failure condition
+	if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_ERROR") == "true" {
+		return details, errors.New("Skipping enrollment details")
+	}
+
+	// If the binary doesn't exist, bail out early.
+	if info, err := os.Stat(osquerydPath); os.IsNotExist(err) {
+		return details, fmt.Errorf("no binary at %s", osquerydPath)
+	} else if info.IsDir() {
+		return details, fmt.Errorf("%s is a directory", osquerydPath)
+	} else if err != nil {
+		return details, fmt.Errorf("statting %s: %w", osquerydPath, err)
+	}
+
+	query := `
+	SELECT
+		osquery_info.version as osquery_version,
+		os_version.build as os_build,
+		os_version.name as os_name,
+		os_version.platform as os_platform,
+		os_version.platform_like as os_platform_like,
+		os_version.version as os_version,
+		system_info.hardware_model,
+		system_info.hardware_serial,
+		system_info.hardware_vendor,
+		system_info.hostname,
+		system_info.uuid as hardware_uuid
+	FROM
+		os_version,
+		system_info,
+		osquery_info;
+
+`
+
+	var respBytes bytes.Buffer
+
+	osq, err := runsimple.NewOsqueryProcess(
+		osquerydPath,
+		runsimple.WithStdout(&respBytes),
+	)
+	if err != nil {
+		return details, fmt.Errorf("create osquery for enrollment details: %w", err)
+	}
+
+	osqCtx, osqCancel := context.WithTimeout(ctx, 5*time.Second)
+	defer osqCancel()
+
+	if err := osq.RunSql(osqCtx, []byte(query)); osqCtx.Err() != nil {
+		return details, fmt.Errorf("query enrollment details context error: %w", osqCtx.Err())
+	} else if err != nil {
+		return details, fmt.Errorf("query enrollment details: %w", err)
+	}
```

Comment:
I think the err being compared here will be the same err as on line 61 since this will be scoped to the if statement

```suggestion
	err = osq.RunSql(osqCtx, []byte(query))
	
	if osqCtx.Err() != nil {
		return details, fmt.Errorf("query enrollment details context error: %w", osqCtx.Err())
	}
	if err != nil {
		return details, fmt.Errorf("query enrollment details: %w", err)
	}
```

Reply:
Testing, it looks like I had worked. (I didn't actually notice this, this is an excellent catch). I renamed the error to make the scoping more clear.